agent/src/control_interface.rs

```diff
@@ -111,14 +135,51 @@ impl PipesChannelContext {
         self.input_pipe_sender.clone()
     }
 
-    pub fn abort_pipes_channel_task(&self) {
+    pub fn abort_control_interface_task(&self) {
         self.task_handle.abort();
     }
 }
 
-impl Drop for PipesChannelContext {
+impl Drop for ControlInterface {
     fn drop(&mut self) {
-        self.abort_pipes_channel_task()
+        self.abort_control_interface_task()
+    }
+}
+
+impl TryFrom<control_interface_info::ControlInterfaceInfo> for ControlInterface {
```

Comment:
This try_from must be tested somehow directly.

Reply:
Removed because the TryFrom implementation cannot be tested, because the ControlInterface::new cannot be intercepted with an context (mixture of mock and not mock).